### PR TITLE
controller: Add cluster backup endpoint

### LINKED ISSUE #2095: Move full cluster backup logic into the controller
Add an endpoint in the controller that acts the same as `flynn cluster backup`.

### PR DESCRIPTION
**Pull request by jvatic**
Closes #2095

### CODE REVIEW

File: cli/cluster.go [hunk 1]
```diff
@@ -337,55 +338,12 @@ func runClusterBackup(args *docopt.Args) error {
...
-			App:       app,
-			Release:   release,
-			Artifact:  artifact,
-			Processes: formation.Processes,
-		}
+	if bar != nil {
+		dest = io.MultiWriter(dest, bar)
 	}
-	if err := tw.WriteJSON("flynn.json", data); err != nil {
+	if err := clusterbackup.RunBackup(client, dest); err != nil {
```
Comment (lmars) [inline cli/cluster.go]: I think the package should be `backup` and the function `Run`, so this becomes `backup.Run`, WDYT?

**Comment by lmars** [comment]
Is it worth adding a test for this? Hit `/backup` and check we get a tar with the relevant files in it?

File: cli/cluster.go (hunk 1 above)
Comment (jvatic) [inline cli/cluster.go]: Yeah, that's simpler.

**Comment by jvatic** [comment]
- [ ] Add integration test

**Comment by jvatic** [comment]
@lmars comments addressed

**Comment by jvatic** [comment]
- [ ] `Content-Disposition` header should be set
- [ ] should be able to use `key` param with this endpoint

**Comment by lmars** [comment]
LGTM

**Pull request merged**